composer/datasets/dataloader.py

```diff
@@ -94,22 +94,42 @@ class DataloaderHparams(hp.Hparams):
     """Hyperparameters to initialize a :class:`~torch.utils.data.Dataloader`.
 
     Parameters:
-        num_workers (int): Number of CPU workers to use per device to fetch data.
-        prefetch_factor (int): Number of samples loaded in advance by each worker.
-            2 means there will be a total of 2 * num_workers samples prefetched across all workers.
-        persistent_workers (bool): Whether or not to shutdown workers after the dataset has been consumed once.
-        pin_memory (bool): Whether or not to copy Tensors into CUDA pinned memory before returning them.
-        timeout (float): Timeout, in seconds, for collecting a batch from workers. Set to 0 for no timeout.
+        num_workers (int, optional): Number of CPU workers to use per device to fetch data.
+            Set to ``0`` to use the main training thread will be used for dataloading.
+            While zero workers can be useful for debugging, it should not be used for performance reasons.
+            (default: ``8``)
+        prefetch_factor (int, optional): Number of samples loaded in advance by each worker.
```

Comment:
Is this samples or batches? I seem to recall it being batches in my past experience with PyTorch, but worth verifying.

Reply:
The docs here say samples: https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader